osal/linux/osal_linux.c

```diff
@@ -418,3 +418,82 @@ void osal_sleep_ms(uint64_t ms)
 {
   usleep(ms * 1000);
 }
+
+/**
+ * @brief Read firmware slot data from file
+ *
+ * @param slotid indicates RUN/UPLOAD/BACKUP slot
+ * @return int 0 for success -1 for failure
+ */
+int osal_read_firmware(uint8_t slotid, void* slot, uint32_t size) {
+  FILE *file = NULL;
+
+  switch(slotid) {
+    case RUN_IMAGE:
```

Comment:
Should the enumeration be in osal.h? It will be required for other platforms too. If csmp_info is a library header, the enumeration should be moved into osal.h

Reply:
I agree, the slots are owned by the osal an consumed by the csmp agent library

Reply:
@ismilak @silabs-ThibautC - thanks for your feedback. Sounds good, I'll move the declaration of xxx_IMAGE enum and _Csmp_Slothdr to osal.h and accordingly update the API signature.